types/reflection/reflection.go

```diff
@@ -0,0 +1,70 @@
+package reflection
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+
+	gogoproto "github.com/cosmos/gogoproto/proto"
+	"golang.org/x/exp/slices"
+	protov2 "google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// GetFileDescriptorSet returns the global file descriptor set by merging
+// the one from gogoproto global registry and from protoregistry.GlobalFiles.
+func GetFileDescriptorSet() (*descriptorpb.FileDescriptorSet, error) {
+	fds := &descriptorpb.FileDescriptorSet{}
+
+	// load gogo proto file descriptors
+	allFds := gogoproto.AllFileDescriptors()
+	haveFileDescriptor := map[string]bool{}
+	for _, compressedBz := range allFds {
+		rdr, err := gzip.NewReader(bytes.NewReader(compressedBz))
+		if err != nil {
+			return nil, err
+		}
+
+		bz, err := io.ReadAll(rdr)
+		if err != nil {
+			return nil, err
+		}
+
+		fd := &descriptorpb.FileDescriptorProto{}
+		err = protov2.Unmarshal(bz, fd)
+		if err != nil {
+			return nil, err
+		}
+
+		// It seems we're registering twice gogo.proto.
+		// See Frojdi's comments in server/grpc/gogoreflection/fix_registration.go.
+		// Skipping here `gogo.proto` and only including `gogoproto/gogo.proto`.
+		if *fd.Name == "gogo.proto" ||
```

Comment:
When logging here, I see both `gogo.proto` and `gogoproto/gogo.proto` being processed. Does it seem correct to you to omit the former like this?